libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -367,7 +402,8 @@ void AP_Baro::update(void)
             if (is_zero(ground_pressure) || isnan(ground_pressure) || isinf(ground_pressure)) {
                 sensors[i].ground_pressure = sensors[i].pressure;
             }
-            float altitude = get_altitude_difference(sensors[i].ground_pressure, sensors[i].pressure);
+            float altitude = get_altitude_difference(sensors[i].ground_pressure, sensors[i].pressure) +
```

Comment:
Very interesting information. Today, I tried a flight, but something was strange. The correction seems to oscillate, what I have never seen with my branch of AC before. I think, this could be a very good explanation for the strange behavior.